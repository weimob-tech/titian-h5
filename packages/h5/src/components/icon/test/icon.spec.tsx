import { newSpecPage } from '@stencil/core/testing';
import { h } from '@stencil/core';
import { TiIcon } from '../index';
import { TestContainer } from '../../common/test/container';

describe('渲染 ti-icon', () => {
  it('基本渲染', async () => {
    const page = await newSpecPage({
      components: [TiIcon, TestContainer],
      template: () => (
        <test-container>
          <ti-icon name="close"></ti-icon>
          <ti-icon name="plus"></ti-icon>
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('图标大小', async () => {
    const page = await newSpecPage({
      components: [TiIcon, TestContainer],
      template: () => (
        <test-container>
          <ti-icon size={20} name="plus"></ti-icon>
          <ti-icon size={40} name="plus"></ti-icon>
          <ti-icon size={60} name="plus"></ti-icon>
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('图标颜色', async () => {
    const page = await newSpecPage({
      components: [TiIcon, TestContainer],
      template: () => (
        <test-container>
          <ti-icon color="red" name="plus"></ti-icon>
          <ti-icon color="blue" name="plus"></ti-icon>
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('图标加载', async () => {
    const page = await newSpecPage({
      components: [TiIcon, TestContainer],
      template: () => (
        <test-container>
          <ti-icon name="close"></ti-icon>
          <ti-icon name="plus"></ti-icon>
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('图标圆角', async () => {
    const page = await newSpecPage({
      components: [TiIcon, TestContainer],
      template: () => (
        <test-container>
          <ti-icon rotate={20} name="plus"></ti-icon>
          <ti-icon rotate={40} name="plus"></ti-icon>
          <ti-icon rotate={60} name="plus"></ti-icon>
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('图标风格', async () => {
    const page = await newSpecPage({
      components: [TiIcon, TestContainer],
      template: () => (
        <test-container>
          <ti-icon name="plus"></ti-icon>
          <ti-icon icon-style="lovely" name="plus"></ti-icon>
          <ti-icon icon-style="popular" name="plus"></ti-icon>
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });
});
