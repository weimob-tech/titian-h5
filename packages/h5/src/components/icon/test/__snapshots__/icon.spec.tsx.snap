// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`渲染 ti-icon 图标加载 1`] = `
<test-container>
  <!---->
  <ti-icon class="titian-icon titian-icon-base titian-icon-close"></ti-icon>
  <ti-icon class="titian-icon titian-icon-base titian-icon-plus"></ti-icon>
</test-container>
`;

exports[`渲染 ti-icon 图标圆角 1`] = `
<test-container>
  <!---->
  <ti-icon class="titian-icon titian-icon-base titian-icon-plus" style="transform: rotate(20deg);"></ti-icon>
  <ti-icon class="titian-icon titian-icon-base titian-icon-plus" style="transform: rotate(40deg);"></ti-icon>
  <ti-icon class="titian-icon titian-icon-base titian-icon-plus" style="transform: rotate(60deg);"></ti-icon>
</test-container>
`;

exports[`渲染 ti-icon 图标大小 1`] = `
<test-container>
  <!---->
  <ti-icon class="titian-icon titian-icon-base titian-icon-plus" style="font-size: 10px;"></ti-icon>
  <ti-icon class="titian-icon titian-icon-base titian-icon-plus" style="font-size: 20px;"></ti-icon>
  <ti-icon class="titian-icon titian-icon-base titian-icon-plus" style="font-size: 30px;"></ti-icon>
</test-container>
`;

exports[`渲染 ti-icon 图标颜色 1`] = `
<test-container>
  <!---->
  <ti-icon class="titian-icon titian-icon-base titian-icon-plus" style="color: red;"></ti-icon>
  <ti-icon class="titian-icon titian-icon-base titian-icon-plus" style="color: blue;"></ti-icon>
</test-container>
`;

exports[`渲染 ti-icon 图标风格 1`] = `
<test-container>
  <!---->
  <ti-icon class="titian-icon titian-icon-base titian-icon-plus"></ti-icon>
  <ti-icon class="titian-icon titian-icon-base titian-icon-plus" icon-style="lovely" style="--icon-family: titian-icon-lovely,titian-icon;"></ti-icon>
  <ti-icon class="titian-icon titian-icon-base titian-icon-plus" icon-style="popular" style="--icon-family: titian-icon-popular,titian-icon;"></ti-icon>
</test-container>
`;

exports[`渲染 ti-icon 基本渲染 1`] = `
<test-container>
  <!---->
  <ti-icon class="titian-icon titian-icon-base titian-icon-close"></ti-icon>
  <ti-icon class="titian-icon titian-icon-base titian-icon-plus"></ti-icon>
</test-container>
`;
